.github/workflows/test.yml

```diff
@@ -29,25 +29,28 @@ jobs:
       - run: npm run report:package
   test_python:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.8', '3.9', '3.10']
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v2
         with:
-          python-version: ${{ env.PYTHON_VERSION }}
+          python-version: ${{ matrix.python }}
       - uses: snok/install-poetry@v1
         with:
-          version: '1.1.13'
+          version: '1.4.1'
           virtualenvs-create: true
           virtualenvs-in-project: true
       - id: poetry-cache
         uses: actions/cache@v2
         with:
           path: .venv
-          key: ${{ runner.os }}-python-${{ env.PYTHON_VERSION }}-${{ hashFiles('**/poetry.lock') }}
+          key: ${{ runner.os }}-python-${{ matrix.python }}-${{ hashFiles('**/poetry.lock') }}
```

Comment:
I think we need an alternative to `${{ matrix.python }}` here.

If we set a cache key containing the `matrix.python` value (e.g. `3.10`), with a poetry env built with `Python==3.10.n`, and the `setup-python` action bumps the Python version to `Python==3.10.n+1`, we may break the pipeline. This is because we will continue to retrieve the cached env built on `3.10.n`.

It seems to me that poetry envs built on the same major.minor Python version should work across bugfix releases, but I don't know much about poetry's internals, and in practice it [seems not to be the case](https://github.com/torchbox/tbxforms/actions/runs/4745633039/jobs/8428375201#step:7:19)

Suggestion:
1. add an `id` entry to the `setup-python` step
2. use `${{ steps.<step_id>.outputs.python-version }}` in the cache key